MediaRecorder.bs

```diff
@@ -569,19 +569,30 @@ dictionary BlobEventInit {
 # Error handling # {#error-handling}
 
 ## General principles ## {#error-handling-principles}
-<em>This section is non-normative.</em>
-
-The UA will throw a {{DOMException}} when the error can be detected at the time
-that the call is made. In all other cases the UA will <a>fire an event</a> named
-{{MediaRecorderErrorEvent}}.  If recording has been started and not yet stopped
-when the error occurs, let <var>blob</var> be the {{Blob}} of collected data so
-far; after raising the error, the UA will <a href="#to-fire-a-blob-event">fire a
-dataavailable event</a> with <var>blob</var>; immediately after the UA will then
-<a>fire an event</a> named <code>stop</code>.
-The UA may set platform-specific limits, such as those for the minimum and
-maximum {{Blob}} size that it will support, or the number of
-{{MediaStreamTrack}}s it will record at once.
-It will signal a fatal error if these limits are exceeded.
+
+If at any moment the UA finds an error condition, the UA MUST queue a task,
+using the DOM manipulation task source, that runs the following steps:
+<ol>
+  <li>Set {{state}} to {{inactive}} and stop gathering data.</li>
+
+  <li>Let <var>blob</var> be the {{Blob}} of collected data so far and
+  let <var>target</var> be the MediaRecorder context object</li>
+
+  <li><a href="#to-fire-a-blob-event">Fire a blob event</a> named
+  <a>dataavailable</a> at <var>target</var> with <var>blob</var>.</li>
+
+  <li><a>Fire an event</a> named <a>stop</a> at <var>target</var></li>
+
+  <li><a>Fire an error event</a> named <a>error</a> with an {{UnknownError}} at
```

Comment:
Question: Do we really want to fire `stop` before `error`? It would make it harder to detect errors IMHO. Most code I've seen seem to treat reaching `stop` without `error` as success, and this would break that assumption.

How about this order: `dataavailable`, `error`, `stop`?